### ISSUE TITLE
How access view component from worker

### ISSUE THREAD

**Issue by mobilemindtec** [NativeScript/NativeScript#3301]
Hi,

I have an app that uses Google Maps. My database has many locations, and according to user's location, I show the locations in a certain radius. Having many locations, the comparison and marker builder process is a bit slow. So I'm trying to implement the logic using workers. My question is:

Can I access an instance of my GoogleMapView from inside the worker? If I pass the GoogleMapView reference instance as message to the worker it doesn't work, both the js instance or native instance..

Can I pass the native component instance to worker?
Can I pass the mapView js instance to worker?

My example:

```
// my view

var mapView, gMap
exports.OnMapReady =  function(args) {

  debug("OnMapReady") 

  mapView = args.object; 
  gMap = mapView.android; 

  var worker = new Worker("../my-worker")
  worker.postMessage({
  	'mapView': mapView,
  	'gMap': gMap
  })
}

// worker 

onmessage = function(message){
	var mapView = message.data.mapView
	var gMap = message.data.gMap

	mapView.addMarker()
	// or 
	gMap.addMarker()
}

```

**Comment by Pip3r4o** [comment]
@mobilemindtec workers are intended to execute only javascript code, without any knowledge of UI elements. That is valid in both Web Workers in your browser and Web Workers in NativeScript. From the provided code sample it becomes clear that you intend to add a marker to the native Map view, that is UI logic that should remain on the main (UI thread). Feel free to perform any calculations that are necessary on the worker thread. 

tl;dr - no, you cannot access your view in a worker.
